packages/connections/src/stores/connections-store.ts

```diff
@@ -141,6 +146,49 @@ async function loadConnections(
   }
 }
 
+function trackConnectionAttemptEvent({ favorite, lastUsed }: ConnectionInfo): void {
+  const trackEvent = {
+    is_favorite: Boolean(favorite),
+    is_recent: Boolean(lastUsed && !favorite),
+    is_new: !lastUsed,
+  };
+  track('Connection Attempt', trackEvent);
+}
+
+async function trackNewConnectionEvent(dataService: DataService, connectionString: string): Promise<void> {
```

Comment:
with the new connection the `dataService` is not be necessary anymore, that same info can be derived from the connection string directly.

Also .. what about username and password, for that the authMechanism may not be necessary and be null, how do we track that?